File: LearnOpencv/2/python/main.py
import cv2
import numpy as np

# 读取彩色图像
image = cv2.imread("C:/github/lib/LearnOpencv/1/c++/cqss.png")
if image is None:
    print("Error: Could not open or find the image!")
    exit()

# 创建一个与原图像相同大小的图像，用于存放结果
result = np.zeros_like(image);

# 遍历每个像素
for y in range(image.shape[0]):
    for x in range(image.shape[1]):
        # 获取当前像素的 RGB 值
        pixel = image[y, x]
        # 计算均值
        average = int((pixel[0] + pixel[1] + pixel[2]) / 3)
        # 设置结果图像的像素为均值
        result[y, x] = [average, average, average]  # OpenCV 使用 BGR 顺序

# 显示原图和处理后的图像
cv2.imshow("Original Image", image)
cv2.imshow("Average Image", result)

# 等待按键
cv2.waitKey(0)

# 关闭窗口
cv2.destroyAllWindows()

File: LearnOpencv/2/c++/main.cpp
#include <iostream>
#include <opencv2\highgui\highgui.hpp>
#include <opencv2\opencv.hpp>
using namespace std;
using namespace cv;

int main()
{
    //读取图像
    Mat img=imread("C:/github/lib/LearnOpencv/1/c++/cqss.png");
    //存放原图像
    Mat img0=img.clone();
    //遍历图像的每一个像素
    for(int h=0;h<img.rows;h++)
    {
        for(int w=0;w<img.cols;w++)
        {
            //计算均值并赋值
            int average=(img.at<Vec3b>(h,w)[0]+img.at<Vec3b>(h,w)[1]+img.at<Vec3b>(h,w)[2])/3;
            img.at<Vec3b>(h,w)=Vec3b(average,average,average);
        }
    }
    //显示图像
    imshow("img0",img0);
    imshow("img",img);
    //等待按键关闭
    waitKey(0);
    return 0;
}